src/structures/ApplicationCommand.js

```diff
@@ -156,6 +156,91 @@ class ApplicationCommand extends Base {
     return this.manager.delete(this, this.guildId);
   }
 
+  /**
+   * Whether this command equals another command. It compares all properties, so for most operations
+   * it is advisable to just compare `command.id === command2.id` as it is much faster and is often
+   * what most users need.
+   * @param {ApplicationCommand|ApplicationCommandData|APIApplicationCommand} command The command to compare with
+   * @returns {boolean}
+   */
+  equals(command) {
+    // If given an id, check if the id matches
+    if (command.id && this.id !== command.id) return false;
+
+    // Check top level parameters
+    const commandType = typeof command.type === 'string' ? command.type : ApplicationCommandTypes[command.type];
+    if (
+      command.name !== this.name ||
+      ('description' in command && command.description !== this.description) ||
+      (commandType && commandType !== this.type) ||
+      // Future proof for options being nullable
+      // TODO: remove ?? 0 on each when nullable
+      (command.options?.length ?? 0) !== (this.options?.length ?? 0) ||
```

Comment:
You don't actually need the `?? 0` because you'd be comparing undefined to undefined if neither one of the commands have options, which is still valid and produces the same output

Reply:
`this.options` always exists currently, the `?.` on the RHS is misleading. I could remove the one on the RHS only but this is accounting for the (yes unlikely) case that someone changed that property in runtime

Reply:
They can also provide an object that they construct in which case it won't have options so it's good to keep both ?. there, just saying that because of your TODO comment, since you can easily remove those